test/Utilities/intervals_test.cc

```diff
@@ -374,48 +324,42 @@ void test_interval_operations() {
 
   static_assert(std::is_same_v<decltype(ct1 * 3.0), microseconds>);
   static_assert(ct1 * 3.0 == 6_us * 3.0);
-  
+
   static_assert(std::is_same_v<decltype(3.0 * ct1), microseconds>);
   static_assert(3.0 * ct1 == 6_us * 3.0);
-  
+
   static_assert(std::is_same_v<decltype(ct1 / 2.0), microseconds>);
   static_assert(ct1 / 2.0 == 6_us / 2.0);
-  
+
   microseconds t1 {    +6.0_us };
   nanoseconds t2  { -4000.0_ns };
 
-  decltype(auto) incr = (t1 += 2000_ns);
-  static_assert(std::is_same_v<decltype(incr), microseconds&>);
-  BOOST_TEST((t1 == 8_us));
-  BOOST_TEST((&incr == &t1));
-  
-  decltype(auto) incr2 = (t1 += t2);
-  static_assert(std::is_same_v<decltype(incr2), microseconds&>);
-  BOOST_TEST((t1 ==  4_us));
-  BOOST_TEST((t2 == -4_us));
-  BOOST_TEST((&incr2 == &t1));
-  
-  decltype(auto) decr = (t1 -= 2000_ns);
-  static_assert(std::is_same_v<decltype(decr), microseconds&>);
-  BOOST_TEST((t1 == 2_us));
-  BOOST_TEST((&decr == &t1));
-  
-  decltype(auto) decr2 = (t1 -= t2);
-  static_assert(std::is_same_v<decltype(decr2), microseconds&>);
-  BOOST_TEST((t1 ==  6_us));
-  BOOST_TEST((t2 == -4_us));
-  BOOST_TEST((&decr2 == &t1));
-  
-  decltype(auto) expand = (t1 *= 2);
-  static_assert(std::is_same_v<decltype(expand), microseconds&>);
-  BOOST_TEST((t1 == 12_us));
-  BOOST_TEST((&expand == &t1));
-  
-  decltype(auto) shrink = (t1 /= 2);
-  static_assert(std::is_same_v<decltype(shrink), microseconds&>);
-  BOOST_TEST((t1 == 6_us));
-  BOOST_TEST((&shrink == &t1));
-  
+  microseconds& incr = (t1 += 2000_ns);
+  BOOST_TEST(t1 == 8_us);
+  BOOST_TEST(&incr == &t1);
+
+  microseconds& incr2 = (t1 += t2);
+  BOOST_TEST(t1 ==  4_us);
+  BOOST_TEST(t2 == -4_us);
+  BOOST_TEST(&incr2 == &t1);
+
+  microseconds& decr = (t1 -= 2000_ns);
+  BOOST_TEST(t1 == 2_us);
+  BOOST_TEST(&decr == &t1);
+
+  microseconds& decr2 = (t1 -= t2);
+  BOOST_TEST(t1 ==  6_us);
+  BOOST_TEST(t2 == -4_us);
+  BOOST_TEST(&decr2 == &t1);
+
+  microseconds& expand = (t1 *= 2);
+  BOOST_TEST(t1 == 12_us);
+  BOOST_TEST(&expand == &t1);
+
+  microseconds& shrink = (t1 /= 2);
+  BOOST_TEST(t1 == 6_us);
+  BOOST_TEST(&shrink == &t1);
+
```

Comment:
The explicit check for the returned type was intentional, because I do not trust conversion-easy C++ to always do the "right" thing.
While the current interface might guarantee a specific return type, I can't exclude that the addition of (potentially unrelated) features would mess this up.

Reply:
You've convinced me.  I've restored the checking, introducing a utility to make the check a little more readable.